### PR TITLE
Update serialization page

### PR DESCRIPTION
**Pull request by slowli**
This PR updates the serialization page according to recent changes to the core (referenced in separate commits).

Related to #154.

### CODE REVIEW

File: src/architecture/serialization.md [hunk 1]
```diff
@@ -180,6 +180,34 @@ little endian.
 `0x01` for true, `0x00` for false. A message with other value stored in place
 of `bool` will not pass validation. Size: 1 byte.
 
+### Floating Point Types
+
+`f32`, `f64`  
+Correspond to floating point types in Rust. The types are stored in little endian
+per `binary32` and `binary64` formats of the IEEE 754 standard. Infinities,
+not-a-number (NaN) values and the negative zero are not supported; they cannot
```
Comment (matklad) [inline src/architecture/serialization.md]: Haven’t we also forbade denormalized numbers?

File: src/architecture/serialization.md [hunk 2]
```diff
@@ -408,11 +433,9 @@ const MY_NEW_MESSAGE_ID: u16 = 1;
 message! {
     struct MessageTwoIntegers {
         const TYPE = MY_NEW_MESSAGE_ID;
-        const ID   = MY_SERVICE_ID;
-        const SIZE = 16;
-
-        field first: u64 [0 => 8]
-        field second: u64 [8 => 16]
+        const ID = MY_SERVICE_ID;
+        first: u64,
```
Comment (matklad) [inline src/architecture/serialization.md]: Micronit: let’s leave a blank line before the fields? I think this is the style which most of messages use, and it seams tidy.

File: src/architecture/serialization.md [hunk 3]
```diff
@@ -474,22 +494,20 @@ let expected_buffer_str = pub_key_str.to_owned() + // Public key
...
 
 | Position | Stored data  | Hexadecimal form | Comment |
 |:--------|:------:|:---------------------|:--------------------------------------------------|
-`0 => 32`  |       | `99 ac e6 c7 21 db 29 3b 0e d5 b4 87 e6 d6 11 1f 22 a8 c5 5d 2a 1b 76 06 b6 fa 6e 6c 29 67 1a a1` | Public key |
-`32  => 36`  | 48    | `30 00 00 00`            | Little endian stored segment pointer, refer to position in data where real string is located |
-`36  => 40`  | 6     | `06 00 00 00`            | Little endian stored segment size |
-`40  => 48` | 1234   | `d2 04 00 00 00 00 00 00`| Number in little endian |
-`48 => 54` | Andrew| `41 6e 64 72 65 77`       | Text bytes in UTF-8 encoding |
+0..32  |       | `99 ac e6 c7 21 db 29 3b 0e d5 b4 87 e6 d6 11 1f 22 a8 c5 5d 2a 1b 76 06 b6 fa 6e 6c 29 67 1a a1` | Public key |
+32..36  | 48    | `30 00 00 00`            | Little endian stored segment pointer, refer to position in data where real string is located |
```
Comment (Maria-Nosyk) [inline src/architecture/serialization.md]: rephrase as "A little endian segment pointer that refers to the string position in the serialization buffer"

File: src/architecture/serialization.md [hunk 4]
```diff
@@ -474,22 +494,20 @@ let expected_buffer_str = pub_key_str.to_owned() + // Public key
...
+36..40  | 6     | `06 00 00 00`            | Little endian stored segment size |
```
Comment (Maria-Nosyk) [inline src/architecture/serialization.md]: "A little endian segment size"

File: src/architecture/serialization.md [hunk 5]
```diff
@@ -474,22 +494,20 @@ let expected_buffer_str = pub_key_str.to_owned() + // Public key
...
+40..48 | 1234   | `d2 04 00 00 00 00 00 00`| Number in little endian |
```
Comment (Maria-Nosyk) [inline src/architecture/serialization.md]: rephrase as "A number in little endian format"

File: src/architecture/serialization.md [hunk 6]
```diff
@@ -474,22 +494,20 @@ let expected_buffer_str = pub_key_str.to_owned() + // Public key
...
+48..54 | Andrew | `41 6e 64 72 65 77`       | Text bytes in UTF-8 encoding |
```
Comment (Maria-Nosyk) [inline src/architecture/serialization.md]: rephrase as "UTF-8 string converted into a byte array"

**Pull request merged**